lib/apimatic-core/utilities/union_type_helper.rb

```diff
@@ -0,0 +1,294 @@
+module CoreLibrary
+  # Helper methods for handling union types.
+  class UnionTypeHelper
+    NONE_MATCHED_ERROR_MESSAGE = 'We could not match any acceptable types against the given JSON.'.freeze
+    MORE_THAN_1_MATCHED_ERROR_MESSAGE = 'There are more than one acceptable type matched against the given JSON.'.freeze
+
+    def self.get_deserialized_value(union_types, value, should_symbolize: false)
```

Comment:
Nits, Non-blocker: The proximity principle should be applied in this class to keep the closely related methods near to each other.

Reply:
done